### PR TITLE
fix: Fix grad norm metric in mcore path

### PR DESCRIPTION
**Pull request by yfw**
# What does this PR do ?

Previously, the grad_norm metric in mcore was added as a microbatch metric and then summed during reporting, meaning the value we were reporting was being multiplied by the number of microbatches. This PR fixes this issue by removing grad_norm from the microbatch metrics and is more consistent with how we report grad norm in the dtensor path.

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

### CODE REVIEW

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 1]
```diff
@@ -1130,9 +1129,7 @@ def train(
             "gpu_name": torch.cuda.get_device_name(),
             "model_dtype": self.dtype,
             "all_mb_metrics": dict(mb_metrics),
-            "grad_norm": torch.tensor(
-                mb_metrics["grad_norm"][-1]
-            ).cpu(),  # TODO @sahilj: return an average or something later
+            "grad_norm": torch.tensor([grad_norm]),
```
Comment (ashors1) [inline nemo_rl/models/policy/megatron_policy_worker.py:1132]: Does this mean we're only returning the last grad norm seen? e.g. if we take multiple train steps per grpo rollout batch, we're ignoring the grad norm from all steps but the last, right?

Dtensor might do this as well; just want to make sure we're aware and okay with that

Comment (yfw) [reply nemo_rl/models/policy/megatron_policy_worker.py:1132]: Yes, good catch. That's correct that we are only taking the last one, but this is consistent with what dtensor is doing.

**Comment by yfw** [comment]
> any chance to add a unit test for this?
> 
> would it be sufficient to test case when using micro_bs=1 mini_bs=2 and micro_bs=2 mini_bs=2 and check grad norm equal?

Added in c951fb9c1ae068fda193175be47c1f933d5346a1
